sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1774,6 +1775,35 @@ class Analyzer(
   }
 
   /**
+   * Substitute Hints.
+   * - BROADCAST/BROADCASTJOIN/MAPJOIN match the closest table with the given name parameters.
```

Comment:
When adding more and more rules, it becomes harder to track the batch-based dependencies. Then, we have to split the batches. Thus, I think we should do our best to document the prerequisite, dependency assumptions in each rule